lib/Map/Cesium/compute3dZoomView.ts

```diff
@@ -0,0 +1,343 @@
+import { isObservableArray } from "mobx";
+import BoundingSphere from "terriajs-cesium/Source/Core/BoundingSphere";
+import Cartesian3 from "terriajs-cesium/Source/Core/Cartesian3";
+import Cartographic from "terriajs-cesium/Source/Core/Cartographic";
+import Ellipsoid from "terriajs-cesium/Source/Core/Ellipsoid";
+import Rectangle from "terriajs-cesium/Source/Core/Rectangle";
+import sampleTerrain from "terriajs-cesium/Source/Core/sampleTerrain";
+import sampleTerrainMostDetailed from "terriajs-cesium/Source/Core/sampleTerrainMostDetailed";
+import BoundingSphereState from "terriajs-cesium/Source/DataSources/BoundingSphereState";
+import DataSource from "terriajs-cesium/Source/DataSources/DataSource";
+import Entity from "terriajs-cesium/Source/DataSources/Entity";
+import Cesium3DTileset from "terriajs-cesium/Source/Scene/Cesium3DTileset";
+import HeightReference from "terriajs-cesium/Source/Scene/HeightReference";
+import filterOutUndefined from "../../Core/filterOutUndefined";
+import pollToPromise from "../../Core/pollToPromise";
+import MappableMixin, {
+  ImageryParts,
+  isCesium3DTileset,
+  isDataSource,
+  isImageryParts,
+  MapItem
+} from "../../ModelMixins/MappableMixin";
+import CameraView from "../../Models/CameraView";
+import Cesium from "../../Models/Cesium";
+import { ZoomTarget } from "../ZoomTarget";
+
+/**
+ * Result of {@link compute3dZoomView} function. Something which can be zoomed to in 3D.
+ *
+ * TODO: Can we can simplify by reducing everything to just CameraView?
+ */
+export type ZoomView3d = BoundingSphere | Cartesian3 | CameraView;
```

Comment:
I think keeping `ZoomView3d` flexible will might help us in the long run - as then it will be easier to implement zoom behaviour for new catalog items/mappable items.

But there is a trade off in therms of maintenance costs. What do you think?